05week/spaceTravelToMars.js

```diff
@@ -1,51 +1,98 @@
 'use strict';
+//require assert
+var assert = require('assert');
 
-let assert = require('assert');
-
-let jobTypes = {
+const jobTypes = {
   pilot: 'MAV',
   mechanic: 'Repair Ship',
   commander: 'Main Ship',
   programmer: 'Any Ship!'
 };
 
-// Your code here
+// test one
+
+class CrewMember {
+  constructor (name, job, specialSkill) {
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+    this.ship = null;
+    this.jobType = jobTypes[this.job];
+    // test two
+    this.enterShip = (ship) => {
+      this.ship = ship;
+      ship.crew.push(this);
+    }
+  }
+}
+// test three
+class Ship {
+  constructor (name, type, ability) {
+    this.name = name;
+    this.type = type;
+    this.ability = ability;
+    this.crew = [];
+    // test four
+    this.missionStatement = ()=> {
+      for (let member of this.crew) {
+        if (member.jobType === this.type) {
+          return this.ability;
+        }
+      }
+      return "Can't perform a mission yet.";
+    }
+  }
+}
+var crewMember1 = new CrewMember('Rick Martinez', 'pilot', 'chemistry');
```

Comment:
we are no longer using var. this would allow the class the ability to be reassigned.